{Release} Upgrade to Azure CLI 2.41.0 (#24138)
Azure/azure-cli: src/azure-cli/HISTORY.rst

* `az appconfig replica`: New command group to support geo-replication (#23747)
* `az appconfig kv export`: Support exporting app configuration settings as references to App Service (#23795)
* `az appconfig kv import`: Ensure app configuration references are not imported from App Service (#23795)
* `az appconfig feature filter update`: Add new command to support updating functionality for feature filters (#24076)

**App Service**

* `az functionapp deployment github-actions`: Add support for linux powershell runtimes (#23939)
* `az functionapp deployment github-actions`: Fix issue where publish profile would not be populated before the github action was run (#23939)
* `az webapp up`: No longer show status during linux deployments (#24051)
* `az webapp deployment source config-zip`: No longer show status during linux deployments (#24051)

**ARM**

* `az deployment group what-if`: Fix an issue where `complete` deployment mode does not work (#23941)

**Backup**

* `az backup policy`: Add support for Smart Tiering policy (#23694)

**Compute**

* [BREAKING CHANGE] `az vmss create`: Update NAT pool to NAT rule V2 for Standard LB SKU when creating VMSS (#23638)

> * `az vm/vmss create`: `--enable-secure-boot` is set to True by default when the `--security-type` used by the VM/VMSS creation is `TrustedLaunch ` (#24004)

Reviewer: ```suggestion
* `az vm/vmss create`: `--enable-secure-boot` is set to True by default when the `--security-type` used by the VM/VMSS creation is `TrustedLaunch` (#24004)
```

* [BREAKING CHANGE] `az network public-ip`: Change `publicIpAddressVersion` to `publicIPAddressVersion` (#24071)
* Fix #23884: `az network application-gateway rule create`: Compatible with v1 SKU (#23907)
* `az network private-endpoint-connection`: Add Provider `Microsoft.AgFoodPlatform/farmBeats` (#23913)
* `az network application-gateway waf-policy managed-rule rule-set`: Support per rule actions in web application firewall (#24054)
* `az network public-ip`: Support ddos protection mode via `--protection-mode` (#24071)

**Packaging**

* Drop Mariner 1.0 RPM package (#24039)

**RDBMS**

* `az mysql flexible-server update`: Expose `--geo-redundant-backup` argument (#23871)
* `az mysql/postgres flexible-server create/update`: Deprecate `Enabled` for `--high-availability` argument (#23847)
* `az mysql flexible-server stop`: Change stopped time logging message (#23979)
* `az mysql flexible-server ad-admin delete`: Disable `aad_auth_only` when dropping AAD admin (#24002)
* `az mysql flexible-server identity remove`: Allow removing all identities in a MySQL server (#24042)

**Reservations**

> * Move commands from azure-cli to azure-cli-extensions (#24097)

Reviewer 2: ```suggestion
* Move commands from azure-cli to `reservation` extension (#24097)
```

Release History
===============

2.41.0
++++++

**ACS**

* [BREAKING CHANGE] `az acs`: Remove the deprecated command group (#23784)

**AD**

* Support special characters in user principal name (#23819)

**AKS**

> * Fix #23779: `az aks install-cli` support to determine the arch of binaries based on system information (#24006)

Reviewer 3: ```suggestion
* Fix #23779: `az aks install-cli`: Support determining the arch of binaries based on system information (#24006)
```

* `az appconfig kv import`: Ensure app configuration references are not imported from App Service (#23795)
* `az appconfig feature filter update`: Add new command to support updating functionality for feature filters (#24076)

**App Service**

* `az functionapp deployment github-actions`: Add support for linux powershell runtimes (#23939)
* `az functionapp deployment github-actions`: Fix issue where publish profile would not be populated before the github action was run (#23939)
* `az webapp up`: No longer show status during linux deployments (#24051)
* `az webapp deployment source config-zip`: No longer show status during linux deployments (#24051)

**ARM**

* `az deployment group what-if`: Fix an issue where `complete` deployment mode does not work (#23941)

**Backup**

* `az backup policy`: Add support for Smart Tiering policy (#23694)

**Compute**

* [BREAKING CHANGE] `az vmss create`: Update NAT pool to NAT rule V2 for Standard LB SKU when creating VMSS (#23638)
* `az vm/vmss create`: `--enable-secure-boot` is set to True by default when the `--security-type` used by the VM/VMSS creation is `TrustedLaunch ` (#24004)

> * `az restore-point create`: Add new parameters `--consistency-mode` to support setting consistency mode (#23789)

Reviewer: ```suggestion
* `az restore-point create`: Add new parameter `--consistency-mode` to support setting consistency mode (#23789)
```

* `az vmss create/update`: Add new parameters `--priority-count` and `--priority-percentage` to support setting priority mix policy (#23786)

> * `az vm/vmss create/update`: Add new parameters `--disk-controller-type` to support setting disk controller type (#23710)

Reviewer: ```suggestion
* `az vm/vmss create/update`: Add new parameter `--disk-controller-type` to support setting disk controller type (#23710)
```

* [BREAKING CHANGE] `az network public-ip`: Change `publicIpAddressVersion` to `publicIPAddressVersion` (#24071)
* Fix #23884: `az network application-gateway rule create`: Compatible with v1 SKU (#23907)
* `az network private-endpoint-connection`: Add Provider `Microsoft.AgFoodPlatform/farmBeats` (#23913)
* `az network application-gateway waf-policy managed-rule rule-set`: Support per rule actions in web application firewall (#24054)
* `az network public-ip`: Support ddos protection mode via `--protection-mode` (#24071)

**Packaging**

* Drop Mariner 1.0 RPM package (#24039)

**RDBMS**

* `az mysql flexible-server update`: Expose `--geo-redundant-backup` argument (#23871)
* `az mysql/postgres flexible-server create/update`: Deprecate `Enabled` for `--high-availability` argument (#23847)
* `az mysql flexible-server stop`: Change stopped time logging message (#23979)
* `az mysql flexible-server ad-admin delete`: Disable `aad_auth_only` when dropping AAD admin (#24002)
* `az mysql flexible-server identity remove`: Allow removing all identities in a MySQL server (#24042)

**Reservations**

> * Remove commands from azure-cli (#24097)

Reviewer 2: ```suggestion
* Move commands from azure-cli to azure-cli-extensions (#24097)
```

**Service Bus**

* [BREAKING CHANGE] `az servicebus namespace create/update`: Remove `--default-action`. Please use `az servicebus namespace network-rule update` command instead (#24092)
* `az servicebus queue/topic create/update`: Support setting max message size (#24092)
* `az servicebus topic subscription create`: Support client affine (#24092)

**Service Connector**

* `az spring-cloud connection create postgres`: Add `--system-identity` for springcloud-postgres connection (#22459)

**SQL**

* `az sql server audit-policy show`: Add `isManagedIdentityInUse` info in output (#23275)

**Storage**

* `az storage blob/container`: Support `--account-name` for non-standard account URL (#23832)

> * `az storage account update`: Fix updating `--default-share-permission` will wipe out other ADProperties issue (#23986)

Reviewer 3: ```suggestion
* `az storage account update`: Fix ADProperties wipe out issue when updating `--default-share-permission` (#23986)
```

* Fix #19311: `az storage remove`: Add support for connection-string (#24049)

> **Synapse**
> 
> * Update Synapse contacts (#23765)
> 

Reviewer 3: ```suggestion
```

> **Synapse**
> 
> * Update Synapse contacts (#23765)
> 

Reviewer 4: ```suggestion
```
